tests/ha/hawk_gui.pm

```diff
@@ -121,6 +124,15 @@ sub run {
 
     # Synchronize with the nodes
     barrier_wait("HAWK_GUI_CHECKED_$cluster_name");
+
+    # Wait for master node to reboot
+    barrier_wait("HAWK_FENCE_$cluster_name");
+
+    # Check that the ex-slave is now the new master
+    sleep 60;
```

Comment:
As I we discussed you may have the case where the "old" master restart quickly and it is promoted master again. Be aware that this can happen.

Reply:
> As I we discussed you may have the case where the "old" master restart quickly and it is promoted master again. Be aware that this can happen.

Can I avoid this inducing sleep in the master?

Reply:
No because pacemaker start before you can have ssh access, so it may works or not, it depends :-)

That's why I never try to implement such a test, and honestly it is not useful at all if the cluster is OK after reboot/fencing of the node. It can be useful to test if you stop node01 and you not restart it. In that case resources hosted on node01 should be migrated to node02 and node02 should be promoted as the new master. But when node01 will be back corosync can move the DC function to node01, there's no real rules about this (in fact there is, you may ask Yan for a detailed explanation I think).

Reply:
> No because pacemaker start before you can have ssh access, so it may works or not, it depends :-)

I would induce the sleep in the master right before reboot, not after.